sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -250,58 +263,77 @@ class ScalaReflectionSuite extends SparkFunSuite {
           "nestedArrayField",
           ArrayType(ArrayType(IntegerType, containsNull = false), containsNull = true)))),
       nullable = true))
+    assert(schema1 === schema2)
   }
 
   test("generic data") {
-    val schema = schemaFor[GenericData[Int]]
-    assert(schema === Schema(
+    val d = GenericData(3)
+    val schema1 = schemaFor[GenericData[Int]]
```

Comment:
@aokolnychyi this test is failing because of type erasure. To be fixed.

Reply:
@dbtsai, friendly reminder, looks like we are waiting for a fix here and next test still.